linkerd/protocol/http/src/main/scala/io/buoyant/linkerd/protocol/http/IstioIdentifier.scala

```diff
@@ -23,14 +24,36 @@ class IstioIdentifier(pfx: Path, baseDtab: () => Dtab, routeCache: RouteCache, c
     }
   }
 
+  def compareStringMatch(headerValue: String, stringMatch: StringMatch): Boolean = {
+    stringMatch.`matchType` match {
+      case Some(OneofMatchType.Exact(value)) => headerValue == value
+      case Some(OneofMatchType.Prefix(pfx)) => headerValue.startsWith(pfx)
+      case Some(OneofMatchType.Regex(r)) => headerValue.matches(r)
```

Comment:
It's too bad we can't compile and store the regex on the StringMatch object.  This is not actionable, I'm just complaining.